libs/core/models/based-params.ts

```diff
@@ -0,0 +1,42 @@
+import { AnimeType } from './anime-type';
+import { Immerable, OmitImmerable } from './immerable';
+
+/** Anime params. */
+export class AnimeParams extends Immerable {
```

Comment:
```suggestion
export class AnimeQuery extends Immerable {
```
It's similar to query